bsp/esp32_s3_korvo_1/esp32_s3_korvo_1_idf5.c

```diff
@@ -8,11 +8,136 @@
 #include "esp_adc/adc_oneshot.h"
 #include "bsp/esp32_s3_korvo_1.h"
 #include "bsp_err_check.h"
+#include "esp_codec_dev_defaults.h"
 
-static const char *TAG = "S3-KORVO-1";
+static const char *TAG = "S3-Korvo-1";
 
+static i2s_chan_handle_t i2s_tx_chan = NULL;
+static i2s_chan_handle_t i2s_rx_chan = NULL;
+static const audio_codec_data_if_t *i2s_data_if_spk = NULL;  /* Codec data interface */
+static const audio_codec_data_if_t *i2s_data_if_mic = NULL;  /* Codec data interface */
 static adc_oneshot_unit_handle_t bsp_adc_handle = NULL;
 
+/* Can be used for i2s_std_gpio_config_t and/or i2s_std_config_t initialization */
+#define BSP_I2S0_GPIO_CFG       \
+    {                          \
+        .mclk = BSP_I2S0_MCLK,  \
+        .bclk = BSP_I2S0_SCLK,  \
+        .ws = BSP_I2S0_LCLK,    \
+        .dout = BSP_I2S0_DOUT,  \
+        .din = BSP_I2S0_DSIN,   \
+        .invert_flags = {      \
+            .mclk_inv = false, \
+            .bclk_inv = false, \
+            .ws_inv = false,   \
+        },                     \
+    }
+
+#define BSP_I2S1_GPIO_CFG       \
+    {                          \
+        .mclk = BSP_I2S1_MCLK,  \
+        .bclk = BSP_I2S1_SCLK,  \
+        .ws = BSP_I2S1_LCLK,    \
+        .dout = BSP_I2S1_DOUT,  \
+        .din = BSP_I2S1_DSIN,   \
+        .invert_flags = {      \
+            .mclk_inv = false, \
+            .bclk_inv = false, \
+            .ws_inv = false,   \
+        },                     \
+    }
+
+/* This configuration is used by default in bsp_audio_init() */
+#define BSP_I2S0_SIMPLEX_MONO_CFG(_sample_rate)                                                         \
+    {                                                                                                 \
+        .clk_cfg = I2S_STD_CLK_DEFAULT_CONFIG(_sample_rate),                                          \
+        .slot_cfg = I2S_STD_PHILIP_SLOT_DEFAULT_CONFIG(I2S_DATA_BIT_WIDTH_16BIT, I2S_SLOT_MODE_MONO), \
+        .gpio_cfg = BSP_I2S0_GPIO_CFG,                                                                 \
+    }
+
+#define BSP_I2S1_CFG(_sample_rate)                                                                      \
+    {                                                                                                 \
+        .clk_cfg = I2S_TDM_CLK_DEFAULT_CONFIG(_sample_rate),                                          \
+        .slot_cfg = I2S_TDM_PHILIP_SLOT_DEFAULT_CONFIG(I2S_DATA_BIT_WIDTH_16BIT, I2S_SLOT_MODE_STEREO, I2S_TDM_SLOT0 | I2S_TDM_SLOT1 | I2S_TDM_SLOT2),\
```

Comment:
This TDM configuration is not usual. Could you provide a docs block explaining the settings?